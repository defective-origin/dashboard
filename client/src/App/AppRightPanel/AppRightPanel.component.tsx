import React from 'react'

// ---| core |---
// ---| pages |---
// ---| screens |---
// ---| components |---

// ---| common |---
import { cn } from 'common/tools'

// ---| self |---
import css from './AppRightPanel.module.scss'

export type AppRightPanelProps = {
  className?: string
  children?: React.ReactNode
}

/**
 * Component description.
 *
 * How to use
 * @example
 * <AppRightPanel />
 */
export function AppRightPanel(props: AppRightPanelProps): JSX.Element {
  const { children, className, ...otherProps } = props
  const _className = cn(css.AppRightPanel, className)

  return <div className={_className} {...otherProps}>RIGHT PANEL{children}</div>
}

AppRightPanel.displayName = 'AppRightPanel'

export default AppRightPanel
