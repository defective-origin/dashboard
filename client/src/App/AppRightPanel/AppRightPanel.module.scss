@import 'src/App/App.module';

:global {
  .dark {
    // --text-color: white;
  }

  .light {
    // --text-color: black;
  }
}

.AppRightPanel {
  border-left: 1px solid var(--divider-color);
  height: 100%;
  box-sizing: border-box;
}
