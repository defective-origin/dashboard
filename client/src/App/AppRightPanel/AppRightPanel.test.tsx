// ---| tests |---
import { render } from 'tests'

// ---| self |---
import AppRightPanel from './AppRightPanel.component'

describe('[AppRightPanel] component', () => {
  it('should render component', () => {
    const container = render(<AppRightPanel />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})
