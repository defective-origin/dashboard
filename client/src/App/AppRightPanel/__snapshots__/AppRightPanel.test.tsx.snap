// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[AppRightPanel] component > should render component 1`] = `
<DocumentFragment>
  <div
    class="_AppRightPanel_59801a"
  >
    RIGHT PANEL
  </div>
</DocumentFragment>
`;
